pyproject.toml

```diff
@@ -95,3 +97,5 @@ skip_glob = ["tests/fixtures/common/collections*"] # Skip ansible content due to
 
     [tool.pylint.format]
     max-line-length = 100
+
+[tool.setuptools_scm]
```

Comment:
This whole dist packaging setup refactoring should go into a separate PR as it's not related to docs. It could be accepted separately while the rest of this PR is still in a state that has to be rejected.

Reply:
It can't yet.  The _version file contains more than just the package version.

Reply:
No, you can do that now. Just don't point it at an existing file. Instead, point it at a gitignored `_scm_version.py` that you'd import from the `_version.py`.
Example: https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/532e343/pyproject.toml#L16 + https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/532e343/src/sphinxcontrib/towncrier/_version.py.

P.S. Also, the same approach is used in `ansible-pylibssh`.

Reply:
Pro tip: create a [`_scm_version.pyi`](https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/532e343/src/sphinxcontrib/towncrier/_scm_version.pyi) to take care of typing.